### PR TITLE
Add the Adjust SDK to Android

### PR DESCRIPTION
**Pull request by mbirghan**
I this PR I added the Adjust SDK to Android. 
Thanks to @strseb for helping :)

Parts of this PR:
- Added a `VPNApplication` to initialize adjust
- Add the Adjust SDK token as a build variable
- Add a way to call adjust functions from C++

The Adjust SDK will only be initialized and track events if the environment variable `ADJUST_SDK_TOKEN` is defined at build time. Also adjust will use the sandbox environment if the application is build for `debug` and will use the production environment when build for `release`.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -144,6 +146,7 @@ android {
         targetSdkVersion 30
         versionCode  System.getenv("VERSIONCODE").toInteger()
         versionName  System.getenv("SHORTVERSION")
+        buildConfigField "String", "ADJUST_SDK_TOKEN", '"' + System.getenv("ADJUST_SDK_TOKEN") + '"'
```
Comment (strseb) [inline android/build.gradle:149]: I mean if anybody does not have the ADJUST_SDK_TOKEN and does a release build this feature will just silently broken. Can we maybe get an assert in the gradle-release task or in the android_package.sh that a release build requires this needs to be set? :)

File: android/src/org/mozilla/firefox/vpn/qt/VPNAdjustHelper.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+package org.mozilla.firefox.vpn.qt;
+
+import com.adjust.sdk.Adjust;
+import com.adjust.sdk.AdjustEvent;
+
+import org.mozilla.firefox.vpn.BuildConfig;
+
+public class VPNAdjustHelper
+{
+    public VPNAdjustHelper() {}
```
Comment (strseb) [inline android/src/org/mozilla/firefox/vpn/qt/VPNAdjustHelper.java:14]: Nobody creats this, lets make the contructor private :)

**Review by strseb — CHANGES_REQUESTED** [review]

File: android/build.gradle (hunk 1 above)
Comment (mbirghan) [reply android/build.gradle:149]: My idea was to make it possible to build for release without having adjust active for testing purposes. But maybe we do not need that.

Comment (strseb) [reply android/build.gradle:149]: we only use the release-job builds for the Playstore so they should be including everything

**Review by strseb — APPROVED** [review]

**Review by bakulf — APPROVED** [review]

**Comment by bakulf** [comment]
Can you add something to the README file about the new env variable?

**Pull request merged**